src/components/manage/Widgets/ArrayWidget.jsx

```diff
@@ -134,54 +128,37 @@ class ArrayWidget extends Component {
     if (
       !this.props.items?.choices?.length &&
       !this.props.choices?.length &&
-      this.vocabBaseUrl
+      this.props.vocabBaseUrl
     ) {
-      this.props.getVocabulary(this.vocabBaseUrl);
+      this.props.getVocabulary(
+        this.props.vocabBaseUrl,
+        null,
+        undefined,
+        100000,
+        this.props.intl.locale,
+      );
     }
+    this.setDefaultValues();
   }
 
-  /**
-   * Initiate search with new query
-   * @param {string} query Search query.
-   * @returns {undefined}
-   */
-  search(query) {
-    if (query.length > 1) {
-      this.props.getVocabulary(this.vocabBaseUrl, query);
-    }
+  componentDidUpdate() {
+    this.setDefaultValues();
   }
 
-  /**
-   * Initiate search with new query
-   * @method loadOptions
-   * @param {string} search Search query.
-   * @param {string} previousOptions The previous options rendered.
-   * @param {string} additional Additional arguments to pass to the next loadOptions.
-   * @returns {undefined}
-   */
-  loadOptions = (search, previousOptions, additional) => {
-    let hasMore = this.props.itemsTotal > previousOptions.length;
-    const offset = this.state.search !== search ? 0 : additional.offset;
-    this.setState({ search });
-
-    if (hasMore || this.state.search !== search) {
-      this.props.getVocabulary(this.vocabBaseUrl, search, offset);
-
-      return {
-        options:
-          intersection(previousOptions, this.props.choices).length ===
-          this.props.choices.length
-            ? []
-            : this.props.choices,
-        hasMore: hasMore,
-        additional: {
-          offset: offset === additional.offset ? offset + 25 : offset,
-        },
-      };
+  setDefaultValues() {
```

Comment:
@giuliaghisini It seems this piece of code is problematic, it leads to infinite updates (it's the Cypress test failng). You can see it in http://localhost:3000/controlpanel/navigation too. In my branch is also failing. Could you please take a look whenever you have time? Thanks!

Reply:
fixed!